.github/workflows/reusable-file-change-check.yaml

```diff
@@ -0,0 +1,32 @@
+name: Pyproject.toml change check
+
+on:
+  workflow_call:
+    inputs:
+      file_path:
+        required: true
+        type: string
+    outputs:
+      changed:
+        description: "Indicates if the file has changed"
+        value: ${{ jobs.check_file_changes.outputs.changed }}
+
+jobs:
+  check_file_changes:
+    runs-on: ubuntu-latest
+    outputs:
+      changed: ${{ steps.check.outputs.changed }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Check file changes
+        id: check
+        run: |
+          echo "checking changes for "${{ inputs.file_path }}"";
+          echo "changed=false" >> $GITHUB_ENV
+          if git diff --name-only HEAD^ "${{ inputs.file_path }}" | grep -q "${{ inputs.file_path }}"; then
```

Comment:
@kirtimanmishrazipstack should we check this against `main` instead of previous HEAD^?

Reply:
@ritwik-g checked against main now